store/helper/helper.go

```diff
@@ -293,22 +300,39 @@ func (h *Helper) FetchRegionTableIndex(metrics map[uint64]RegionMetric, allSchem
 			logutil.BgLogger().Error("locate region failed", zap.Error(err))
 			continue
 		}
+		regionsInfo := []*RegionInfo{
+			{
+				ID:       int64(regionID),
+				StartKey: bytesKeyToEncodeHex(region.StartKey),
+				EndKey:   bytesKeyToEncodeHex(region.EndKey),
+			}}
+
+		regionsTableInfos := h.ParseRegionsTableInfos(regionsInfo, tables)
+
+		if tableInfos, ok := regionsTableInfos[int64(regionID)]; ok {
+			if len(tableInfos) == 0 {
+				hotTables = append(hotTables, t)
+				continue
+			}
+			for _, tableInfo := range tableInfos {
+				t.DbName = tableInfo.DB.Name.String()
+				t.TableName = tableInfo.Table.Name.O
+				t.TableID = tableInfo.Table.ID
+				if tableInfo.IsIndex {
+					t.IndexName = tableInfo.Index.Name.O
+					t.IndexID = tableInfo.Index.ID
+				} else {
+					t.IndexName = ""
+					t.IndexID = 0
+				}
 
-		hotRange, err := NewRegionFrameRange(region)
-		if err != nil {
-			return nil, err
-		}
-		f := h.FindTableIndexOfRegion(allSchemas, hotRange)
-		if f != nil {
-			t.DbName = f.DBName
-			t.TableName = f.TableName
-			t.TableID = f.TableID
-			t.IndexName = f.IndexName
-			t.IndexID = f.IndexID
+				hotTables = append(hotTables, t)
+			}
+		} else {
+			hotTables = append(hotTables, t)
```

Comment:
`ok is false` goes into this branch, which means the hot region's schema is unfound due to reasons like schma change. I think ignore this kind of  hot regions is a better handle. So I remove it in new commit and update `TestHotRegion` to avoid this case to pass the test.

Reply:
@nolouch , ptal